### ISSUE TITLE
Cashu V1 Endpoints

### ISSUE THREAD

**Issue by Kodylow** [Kodylow/fedimint-http#9]
The router should have cashu compatible endpoints per the V1 Nut specifications

Https://GitHub.com/cashubtc/nuts

I've already set up most of the structure in the cashu folder, just needs to get finished and polished for mint, melt, validate, etc

For cashu encoding the ecash strings: fedimint currently uses a binary encoding that's hard to parse, all the cashu endpoints should use the url safe base64 encodings per the cashu spec.

 Everything in the fedimint ecash encoding can map to cashu, only differences are that fedimint uses a federation prefix id that becomes the mint url and the notes need to be structured with the amount in the note itself instead of grouped by amount which we do in TieredMulti<SpendableNote> .

**Comment by richarddushime** [comment]
@Kodylow I would love to work on this issue 
Thank you
